src/Compilers/CSharp/Test/Emit2/Semantics/CollectionLiteralTests.cs

```diff
@@ -5644,6 +5667,2548 @@ private static void VerifyTypes(SemanticModel model, ExpressionSyntax expr, stri
+                using System.Runtime.CompilerServices;
+                public class Container<T>
+                {
+                    [CollectionBuilder(typeof(MyCollectionBuilder), "Create")]
+                    public struct MyCollection : IEnumerable<T>
+                    {
+                        private readonly List<T> _list;
+                        public MyCollection(List<T> list) { _list = list; }
+                        public IEnumerator<T> GetEnumerator() => _list.GetEnumerator();
+                        IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+                    }
+                    public sealed class MyCollectionBuilder
+                    {
+                        public static MyCollection Create(ReadOnlySpan<T> items)
+                            => new MyCollection(new List<T>(items.ToArray()));
+                    }
+                }
+                #pragma warning disable 219
+                class Program
+                {
+                    static void Main()
+                    {
+                        Container<string>.MyCollection x = [];
+                        Container<int>.MyCollection y = [default];
+                        Container<object>.MyCollection z = new();
+                    }
+                }
+                """;
+            var comp = CreateCompilation(new[] { source, s_constructibleCollectionAttributeDefinition }, targetFramework: TargetFramework.Net70);
+            comp.VerifyEmitDiagnostics(
+                // 0.cs(7,24): error CS0416: 'Container<T>.MyCollectionBuilder': an attribute argument cannot use type parameters
+                //     [CollectionBuilder(typeof(MyCollectionBuilder), "Create")]
+                Diagnostic(ErrorCode.ERR_AttrArgWithTypeVars, "typeof(MyCollectionBuilder)").WithArguments("Container<T>.MyCollectionBuilder").WithLocation(7, 24),
+                // 0.cs(27,42): error CS1061: 'Container<int>.MyCollection' does not contain a definition for 'Add' and no accessible extension method 'Add' accepting a first argument of type 'Container<int>.MyCollection' could be found (are you missing a using directive or an assembly reference?)
+                //         Container<int>.MyCollection y = [default];
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "default").WithArguments("Container<int>.MyCollection", "Add").WithLocation(27, 42));
+        }
+
+        [CombinatorialData]
+        [ConditionalTheory(typeof(CoreClrOnly))]
+        public void CollectionBuilder_GenericType_04(bool useCompilationReference)
+        {
+            string sourceA = """
+                using System;
+                using System.Collections;
+                using System.Collections.Generic;
+                using System.Runtime.CompilerServices;
+                public class Container<T>
+                {
+                    [CollectionBuilder(typeof(MyCollectionBuilder), "Create")]
```

Comment:
do we have a test where teh attribute is on the collection, but points to a nested builder type within that collection?

Reply:
There was one test, but it was an inaccessible builder. Added `CollectionBuilder_NestedBuilder`.